### ISSUE TITLE
Upgrade maven plugin jackson version to 2.17.2

### ISSUE THREAD

**Issue by takanuva15** [victools/jsonschema-generator#481]
Hi, I'm working on upgrading our dependencies, and it seems the maven plugin [imports](https://mvnrepository.com/artifact/com.github.victools/jsonschema-maven-plugin/4.36.0) jackson 2.14. Unfortunately, that doesn't seem to be compatible with jackson 2.17, which causes this error to trigger for me:
```
Caused by: org.apache.maven.plugin.PluginContainerException: An API incompatibility was encountered while executing com.github.victools:jsonschema-maven-plugin:4.36.0:generate: java.lang.NoSuchMethodError: 'void com.fasterxml.jackson.core.base.GeneratorBase.<init>(int, com.fasterxml.jackson.core.ObjectCodec, com.fasterxml.jackson.core.io.IOContext)'
```

Another poster on a different repo faced a [similar issue](https://github.com/OpenAPITools/openapi-generator/issues/18753#issuecomment-2139133841) and mentioned it's due to the conflict between jackson 2.14 and 2.17. I saw you [merged a PR](https://github.com/victools/jsonschema-generator/pull/480) to update dependencies 4 days ago - is that the only thing we need to release to upgrade the jackson version and fix this issue?

---

EDIT: As a temporary patch, I force-upgraded the plugin dependency's jackson-core version to 2.17 and it seems to be working:
```xml
<plugin>
    <groupId>com.github.victools</groupId>
    <artifactId>jsonschema-maven-plugin</artifactId>
    <version>4.36.0</version>
    ...
    <dependencies>
        <dependency>
            <groupId>com.fasterxml.jackson.core</groupId>
            <artifactId>jackson-core</artifactId>
            <version>2.17.2</version>
        </dependency>
    </dependencies>
</plugin>
```

**Comment by CarstenWickner** [comment]
#480 would not have solved this, but in #490 I did move the jackson version to `2.17.2`.
That achieves the same as your workaround, yes.

**Comment by CarstenWickner** [comment]
Release [v4.37.0](https://github.com/victools/jsonschema-generator/releases/tag/v4.37.0) has just been published containing the dependency update mentioned above.
